backend/src/utils/resourceUtils.ts

```diff
@@ -80,60 +76,76 @@ const fetchInstalledKfdefs = async (fastify: KubeFastifyInstance): Promise<KfDef
   return kfdef?.spec?.applications || [];
 };
 
-const fetchApplicationDefs = (): Promise<OdhApplication[]> => {
-  const normalizedPath = path.join(__dirname, '../../../data/applications');
-  const applicationDefs: OdhApplication[] = [];
+const fetchOdhApplications = async (fastify: KubeFastifyInstance): Promise<OdhApplication[]> => {
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  const namespace = fastify.kube.namespace;
   const featureFlags = getComponentFeatureFlags();
-  fs.readdirSync(normalizedPath).forEach((file) => {
-    if (yamlRegExp.test(file)) {
-      try {
-        const doc = jsYaml.load(fs.readFileSync(path.join(normalizedPath, file), 'utf8'));
-        if (!doc.spec.featureFlag || featureFlags[doc.spec.featureFlag]) {
-          applicationDefs.push(doc);
-        }
-      } catch (e) {
-        console.error(`Error loading application definition ${file}: ${e}`);
+
+  let odhApplications: OdhApplication[];
+  try {
+    const res = await customObjectsApi.listNamespacedCustomObject(
+      'applications.console.openshift.io',
+      'v1alpha1',
+      namespace,
+      'odhapplications',
+    );
+    const cas = (res?.body as { items: OdhApplication[] })?.items;
+    odhApplications = cas.reduce((acc, ca) => {
+      if (!ca.spec.featureFlag || featureFlags[ca.spec.featureFlag]) {
+        acc.push(ca);
       }
-    }
-  });
-  return Promise.resolve(applicationDefs);
+      return acc;
+    }, []);
+  } catch (e) {
+    fastify.log.error(e, 'failed to get odhapplications');
+    const error = createError(500, 'failed to get odhapplications');
+    error.explicitInternalServerError = true;
+    error.error = 'failed to get odhapplications';
+    error.message =
+      'Unable to get OdhApplication resources. Please ensure the Open Data Hub operator has been installed.';
+    throw error;
+  }
+  return Promise.resolve(odhApplications);
 };
 
-const fetchDocs = async (): Promise<OdhDocument[]> => {
-  const normalizedPath = path.join(__dirname, '../../../data/docs');
-  const docs: OdhDocument[] = [];
+const fetchOdhDocuments = async (fastify: KubeFastifyInstance): Promise<OdhDocument[]> => {
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  const namespace = fastify.kube.namespace;
   const featureFlags = getComponentFeatureFlags();
-  const appDefs = await fetchApplicationDefs();
-
-  fs.readdirSync(normalizedPath).forEach((file) => {
-    if (yamlRegExp.test(file)) {
-      try {
-        const doc: OdhDocument = jsYaml.load(
-          fs.readFileSync(path.join(normalizedPath, file), 'utf8'),
-        );
-        if (doc.spec.featureFlag) {
-          if (featureFlags[doc.spec.featureFlag]) {
-            docs.push(doc);
-          }
-          return;
-        }
-        if (!doc.spec.appName || appDefs.find((def) => def.metadata.name === doc.spec.appName)) {
```

Comment:
This prevents adding docs for applications that are feature flagged out. I think we still need to handle this.

Reply:
added this logic